builtin/read_osh.py

```diff
@@ -174,6 +176,16 @@ def _ReadPortion(delim_byte, max_chars, cmd_ev):
             eof = True
             break
 
+        elif backslash:
+            backslash = False
+            if ch == delim_byte:
+                continue
```

Comment:
Sorry, I now checked the behavior of Bash again and realized that Bash's behavior is slightly different from this.

Reply:
Bash seems to swallow "backslash + newline" (instead of "backslash + delimiter") regardless of `delim_byte`. I pushed a fix 54332fb76.